src/updatechecker.cpp

```diff
@@ -234,13 +234,13 @@ void UpdateChecker::Run()
 
         if (checkUpdates)
         {
-            const time_t currentTime = time(NULL);
-            time_t lastCheck = currentTime;
```

Comment:
Isn't the simplest possible fix to just do `time_t lastCheck = 0` here instead of your larger diff? Am I missing something (again)?